framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3303,8 +3303,7 @@ protected boolean isAccessAllowed(
             return true;
         }
 
-        String when =
-                AnnotationUtils.getElementValueClassName(unused, "when", false).toString().intern();
+        String when = AnnotationUtils.getElementValueClassName(unused, "when", false).toString();
```

Comment:
How about replacing these uses of `String` by `Name`? I see a few places where we get a `Name` from the compiler, call `toString().intern()` and then pass around the `String`.
We're already changing APIs in this release, so we might as well clean this up.

Reply:
I tried this, but didn't see any speed up.  (`containsSameByName` isn't one of the top 20 methods.)  I also tried comparing by `Name` in `areSameIgnoringValues`, but also didn't see any speed up.

Reply:
Even without an immediate speed-up, wouldn't it be the nicer API? The compiler already uses `Name` everywhere, so why convert to `String`.
If you've already done the work, why not commit it?

Reply:
I changed some of the code, but I didn't update the documentation or make the change in every place possible. To really make this change, every `SameByName` method needs to be duplicated, because there are still places where only we have a String but not a Name.  Since it doesn't help with performance, this change is at the bottom of my list.  (And shouldn't hold up this pull request.)

Reply:
I don't think you can make a fair assessment of performance if you didn't convert all locations.
I wouldn't duplicate all these methods. The places that use `String`s should use `com.sun.tools.javac.util.Names.fromString(String)` to generate a `Name`. If that only happens in few locations, we would safe those String instantiations and comparisons.
But I'm fine with this PR if it gives us a large performance benefit.